rpc/transport/transport.go

```diff
@@ -96,6 +107,18 @@ func NewPackedStream(rwc io.ReadWriteCloser) Transport {
 	return New(newStreamCodec(rwc, packedEncoding{}))
 }
 
+func (t *transport) allocArena() *capnp.MultiSegmentArena {
+	return t.arenaPool.Get().(*capnp.MultiSegmentArena)
+}
+
+func (t *transport) freeArena(arena *capnp.MultiSegmentArena) {
+	// Truncate each segment:
+	for i := 0; i < len(*arena); i++ {
```

Comment:
Yeah, moving the message pooling to the package level is something I've thought about and might be a good idea.

Reply:
I'm strongly in favor of this, FWIW.  The only downside I can think of is that it might be possible for malicious code to read data from a freed `Arena`, though we could zero out the segments if that becomes a practical concern.